src/govuk/components/skip-link/skip-link.js

```diff
@@ -4,90 +4,91 @@ import '../../vendor/polyfills/Event' // addEventListener and event.target norma
 
 function SkipLink ($module) {
   this.$module = $module
+  this.$linkedElement = null
+  this.linkedElementListener = false
 }
 
 /**
  * Initialise the component
  */
 SkipLink.prototype.init = function () {
-  var $module = this.$module
   // Check for module
-  if (!$module) {
+  if (!this.$module) {
     return
   }
 
-  $module.addEventListener('click', this.handleClick.bind(this))
+  // Check for linked element
+  this.$linkedElement = this.getLinkedElement()
+  if (!this.$linkedElement) {
+    return
+  }
+
+  this.$module.addEventListener('click', this.focusLinkedElement.bind(this))
 }
 
 /**
-* Click event handler
+* Get linked element
 *
-* @param {MouseEvent} event - Click event
+* @returns {HTMLElement} $linkedElement - DOM element linked to from the skip link
 */
-SkipLink.prototype.handleClick = function (event) {
-  var target = event.target
+SkipLink.prototype.getLinkedElement = function () {
+  var linkedElementId = this.getFragmentFromUrl()
 
-  if (this.focusLinkedElement(target)) {
-    event.preventDefault()
+  if (!linkedElementId) {
+    return false
   }
+
+  return document.getElementById(linkedElementId)
 }
 
 /**
  * Focus the linked element
  *
- * @param {HTMLElement} $target - Event target
- * @returns {boolean} True if the linked element was able to be focussed
+ * Set tabindex and helper CSS class. Set listener to remove them on blur.
  */
-SkipLink.prototype.focusLinkedElement = function ($target) {
-  // If the element that was clicked does not have a href, return early
-  if ($target.href === false) {
-    return false
-  }
-
-  var linkedElementId = this.getFragmentFromUrl($target.href)
-  var $linkedElement = document.getElementById(linkedElementId)
-  if (!$linkedElement) {
-    return false
-  }
+SkipLink.prototype.focusLinkedElement = function () {
+  var $linkedElement = this.$linkedElement
 
   if (!$linkedElement.getAttribute('tabindex')) {
-    // Set the content tabindex to -1 so it can be focused with JavaScript.
+    // Set the element tabindex to -1 so it can be focused with JavaScript.
     $linkedElement.setAttribute('tabindex', '-1')
     $linkedElement.classList.add('govuk-skip-link-focused-element')
 
-    $linkedElement.addEventListener('blur', this.removeFocusProperties.bind(this, $linkedElement))
+    // Add listener for blur on the focused element (unless the listener has previously been added)
+    if (!this.linkedElementListener) {
+      this.$linkedElement.addEventListener('blur', this.removeFocusProperties.bind(this))
+      this.linkedElementFocused = true
```

Comment:
If I understand the intention correctly, I think this needs to be `linkedElementListener` for the guard to work?

```suggestion
      this.linkedElementListener = true
```

Reply:
Thanks for the eagle eye, I kept fluctuating on the variable name and this slipped through.